src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -1654,30 +1654,41 @@ namespace winrt::Microsoft::Terminal::Control::implementation
     // - text: the text to search
     // - goForward: boolean that represents if the current search direction is forward
     // - caseSensitive: boolean that represents if the current search is case sensitive
+    // - resetOnly: If true, only Reset() will be called, if anything. FindNext() will never be called.

```

Comment:
```suggestion
    // - resetOnly: If true, only Search::IsStale() will be called, if anything. FindNext() will never be called.
```

right?

Reply:
No, actually the comment is correct, but the boolean logic inside the function is very difficult to read. Basically, if `resetOnly` is true then `if (searchInvalidated || !resetOnly)` is only entered if `searchInvalidated` is true. If `searchInvalidated` is true then only `Reset()` will be called.

Reply:
oh yea that's hard to parse